pkg/gameservers/health.go

```diff
@@ -125,8 +125,11 @@ func (hc *HealthController) unschedulableWithNoFreePorts(pod *corev1.Pod) bool {
 func (hc *HealthController) failedContainer(pod *corev1.Pod) bool {
 	container := pod.Annotations[agonesv1.GameServerContainerAnnotation]
 	for _, cs := range pod.Status.ContainerStatuses {
-		if cs.Name == container && cs.State.Terminated != nil {
-			return true
+		if cs.Name == container {
+			// sometimes on a restart, these cs.State can be running and the last state will be merged
+			if cs.State.Terminated != nil || cs.LastTerminationState.Terminated != nil {
```

Comment:
nit: can probably be `return cs.State.Terminated != nil || cs.LastTerminationState.Terminated != nil` to break the loop once the desired container is found.

Reply:
Yes! Change made.